modules/services/indexing/src/main/java/org/fao/geonet/indexing/service/IndexingService.java

```diff
@@ -202,37 +202,43 @@ protected static String collectDbProperties(AbstractMetadata r) {
 
 
   private BulkRequest buildBulkRequest(IndexRecords indexRecords) {
-    BulkRequest bulkRequest = new BulkRequest(index);
+    BulkRequest.Builder requestBuilder = new BulkRequest.Builder()
+        .index(index)
+        .refresh(Refresh.True);
     ObjectMapper mapper = new ObjectMapper();
 
     indexRecords.getIndexRecord().forEach(r -> {
       try {
-        IndexRequest indexRequest = new IndexRequest();
-        indexRequest.id(r.getId());
-        indexRequest.source(mapper.writeValueAsString(r), XContentType.JSON);
-        bulkRequest.add(indexRequest);
+        String json = mapper.writeValueAsString(r);
+        requestBuilder.operations(op -> op
+            .index(idx -> idx
+                .index(index)
+                .id(r.getId())
+                .document(json)
+            )
+        );
       } catch (JsonProcessingException jsonProcessingException) {
         jsonProcessingException.printStackTrace();
       }
     });
-    return bulkRequest;
+    return requestBuilder.build();
   }
 
   private void sendToIndex(IndexRecords indexRecords,
       IndexingReport report) {
     BulkRequest bulkRequest = buildBulkRequest(indexRecords);
     try {
       // TODO: Asynchronous?
-      BulkResponse bulkItemResponses = client.bulk(bulkRequest, RequestOptions.DEFAULT);
+      BulkResponse bulkItemResponses = client.bulk(bulkRequest);
```

Comment:
Should be catch also `ElasticsearchException` below?

Reply:
Added.